src/server/transaction.cc

```diff
@@ -94,12 +94,36 @@ bool CheckLocks(const DbSlice& db_slice, IntentLock::Mode mode, const KeyLockArg
 struct ScheduleContext {
   Transaction* trans;
   bool optimistic_execution = false;
+
+  std::atomic<ScheduleContext*> next{nullptr};
+
   std::atomic_uint32_t fail_cnt{0};
 
   ScheduleContext(Transaction* t, bool optimistic) : trans(t), optimistic_execution(optimistic) {
   }
 };
 
+struct ScheduleQ {
+  base::MPSCIntrusiveQueue<ScheduleContext> queue;
+
+  static constexpr size_t kSz = sizeof(queue);
+
+  char pad1[64];
+  atomic_bool armed{false};
+  char pad2[60];
+};
```

Comment:
🤓 

```c++
alignas(64) MPSC queue;
alignas(64) atomic_bool armed;
```

Reply:
but isntead of 64 exist better option std::hardware_destructive_interference_size

Reply:
yes, but this is just an awful name that translated to 64 anyways

Reply:
in the future it can be 128 so awful name is better than const

Reply:
you are right, my answer was emotional 😺